src/wasm-interpreter.h

```diff
@@ -1898,6 +1899,23 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
         }
         return makeGCData(contents, curr->type);
       }
+      case StringNewFromCodePoint: {
+        uint32_t codePoint = ptr.getSingleValue().geti32();
+        std::string str;
+        str += char(codePoint & 0xff);
+        str += char((codePoint >> 8) & 0xff);
+        if (codePoint > 0xffff) {
+          str += char((codePoint >> 16) & 0xff);
+          str += char((codePoint >> 24) & 0xff);
+        }
+        std::stringstream null;
+        if (!String::convertUTF16ToUTF8(null, str)) {
+          trap("invalid code point");
+        }
```

Comment:
This check for invalid code points is probably not behaving correctly because this is not the correct way to encode a code point as UTF-16.

To check for validity, all you have to do is check [`codePoint <= 0x10FFFF`](https://github.com/WebAssembly/js-string-builtins/blob/main/proposals/js-string-builtins/Overview.md#wasmjs-string-fromcodepoint). After that, you can use `String::writeWTF16CodePoint` to get the proper encoding, then you can convert that from a string to an array literal.

Reply:
Thanks! I am still very fuzzy on the UTF stuff :smile: I thought I just need to emit two 16-bit chunks here...